[DBG_X] Compilation failure due to signed/unsigned comparision in SiPixelDigisClustersFromSoAT

Compilation failure in IB CMSSW_13_3_DBG_X_2023-07-20-2300 ([full log](https://cmssdt.cern.ch/SDT/cgi-bin/buildlogs/el8_amd64_gcc11/CMSSW_13_3_DBG_X_2023-07-20-2300/RecoLocalTracker/SiPixelClusterizer)):
```
/data/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/9a03ff54ef53dbd2dc0132c33064f253/opt/cmssw/el8_amd64_gcc11/cms/cmssw/CMSSW_13_3_DBG_X_2023-07-20-2300/src/RecoLocalTracker/SiPixelClusterizer/plugins/SiPixelDigisClustersFromSoA.cc: In instantiation of 'void SiPixelDigisClustersFromSoAT<TrackerTraits>::produce(edm::StreamID, edm::Event&, const edm::EventSetup&) const [with TrackerTraits = pixelTopology::HIonPhase1]':
/data/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/9a03ff54ef53dbd2dc0132c33064f253/opt/cmssw/el8_amd64_gcc11/cms/cmssw/CMSSW_13_3_DBG_X_2023-07-20-2300/src/RecoLocalTracker/SiPixelClusterizer/plugins/SiPixelDigisClustersFromSoA.cc:74:6:   required from here
  /data/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/9a03ff54ef53dbd2dc0132c33064f253/opt/cmssw/el8_amd64_gcc11/cms/cmssw/CMSSW_13_3_DBG_X_2023-07-20-2300/src/RecoLocalTracker/SiPixelClusterizer/plugins/SiPixelDigisClustersFromSoA.cc:186:26: error: comparison of integer expressions of different signedness: 'int32_t' {aka 'int'} and 'const uint32_t' {aka 'const unsigned int'} [-Werror=sign-compare]
   186 |     assert(digis.clus(i) < TrackerTraits::maxNumClustersPerModules);
      |            ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/data/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/9a03ff54ef53dbd2dc0132c33064f253/opt/cmssw/el8_amd64_gcc11/cms/cmssw/CMSSW_13_3_DBG_X_2023-07-20-2300/src/RecoLocalTracker/SiPixelClusterizer/plugins/SiPixelDigisClustersFromSoA.cc: In instantiation of 'void SiPixelDigisClustersFromSoAT<TrackerTraits>::produce(edm::StreamID, edm::Event&, const edm::EventSetup&) const [with TrackerTraits = pixelTopology::Phase2]':
/data/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/9a03ff54ef53dbd2dc0132c33064f253/opt/cmssw/el8_amd64_gcc11/cms/cmssw/CMSSW_13_3_DBG_X_2023-07-20-2300/src/RecoLocalTracker/SiPixelClusterizer/plugins/SiPixelDigisClustersFromSoA.cc:74:6:   required from here
  /data/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/9a03ff54ef53dbd2dc0132c33064f253/opt/cmssw/el8_amd64_gcc11/cms/cmssw/CMSSW_13_3_DBG_X_2023-07-20-2300/src/RecoLocalTracker/SiPixelClusterizer/plugins/SiPixelDigisClustersFromSoA.cc:186:26: error: comparison of integer expressions of different signedness: 'int32_t' {aka 'int'} and 'const uint32_t' {aka 'const unsigned int'} [-Werror=sign-compare]
 /data/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/9a03ff54ef53dbd2dc0132c33064f253/opt/cmssw/el8_amd64_gcc11/cms/cmssw/CMSSW_13_3_DBG_X_2023-07-20-2300/src/RecoLocalTracker/SiPixelClusterizer/plugins/SiPixelDigisClustersFromSoA.cc: In instantiation of 'void SiPixelDigisClustersFromSoAT<TrackerTraits>::produce(edm::StreamID, edm::Event&, const edm::EventSetup&) const [with TrackerTraits = pixelTopology::Phase1]':
/data/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/9a03ff54ef53dbd2dc0132c33064f253/opt/cmssw/el8_amd64_gcc11/cms/cmssw/CMSSW_13_3_DBG_X_2023-07-20-2300/src/RecoLocalTracker/SiPixelClusterizer/plugins/SiPixelDigisClustersFromSoA.cc:74:6:   required from here
  /data/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/9a03ff54ef53dbd2dc0132c33064f253/opt/cmssw/el8_amd64_gcc11/cms/cmssw/CMSSW_13_3_DBG_X_2023-07-20-2300/src/RecoLocalTracker/SiPixelClusterizer/plugins/SiPixelDigisClustersFromSoA.cc:186:26: error: comparison of integer expressions of different signedness: 'int32_t' {aka 'int'} and 'const uint32_t' {aka 'const unsigned int'} [-Werror=sign-compare]
 cc1plus: some warnings being treated as errors
```

Pinging @AdrianoDee as the author of https://github.com/cms-sw/cmssw/pull/41632

--- a/RecoLocalTracker/SiPixelClusterizer/plugins/SiPixelDigisClustersFromSoA.cc
+++ b/RecoLocalTracker/SiPixelClusterizer/plugins/SiPixelDigisClustersFromSoA.cc
@@ -183,7 +183,7 @@ void SiPixelDigisClustersFromSoAT<TrackerTraits>::produce(edm::StreamID,
       // fill clusters
 #ifdef EDM_ML_DEBUG
     assert(digis.clus(i) >= 0);
-    assert(digis.clus(i) < TrackerTraits::maxNumClustersPerModules);
+    assert(digis.clus(i) < static_cast<int32_t>(TrackerTraits::maxNumClustersPerModules));
 #endif
     nclus = std::max(digis.clus(i), nclus);
     auto row = dig.row();

